compiler/src/dotty/tools/dotc/transform/RepeatableAnnotations.scala

```diff
@@ -0,0 +1,49 @@
+package dotty.tools.dotc
+package transform
+
+import core._
+import ast.tpd._
+import Contexts._
+import MegaPhase._
+import Annotations._
+import Symbols.defn
+import Constants._
+import Types._
+import Decorators._
+
+class RepeatableAnnotations extends MiniPhase:
+  override def phaseName = "repeatableAnnotations"
+
+  override def transformTypeDef(tree: TypeDef)(using Context): Tree = transformDef(tree)
+  override def transformValDef(tree: ValDef)(using Context): Tree = transformDef(tree)
+  override def transformDefDef(tree: DefDef)(using Context): Tree = transformDef(tree)
+
+  private def transformDef(tree: DefTree)(using Context) =
+    val annotations = tree.symbol.annotations
+    if (!annotations.isEmpty) then
+      tree.symbol.annotations = aggregateAnnotations(tree.symbol.annotations)
+    tree
+
+  private def aggregateAnnotations(annotations: Seq[Annotation])(using Context): List[Annotation] =
+    val annsByType = annotations.groupBy(_.symbol)
+    annsByType.flatMap {
+      case (_, a :: Nil) => a :: Nil
+      case (sym, anns) if sym.derivesFrom(defn.ClassfileAnnotationClass) =>
+        sym.annotations.find(_ matches defn.JavaRepeatableAnnot).flatMap(_.argumentConstant(0)) match
```

Comment:
```suggestion
        sym.annotations.getAnnotation(defn.JavaRepeatableAnnot).flatMap(_.argumentConstant(0)) match
```

Reply:
Minor: I'd pattern match on `sym.annotations.getAnnotation(defn.JavaRepeatableAnnot)`, and then call `argumentConstant(0)` in the `Some` branch. This is more clear.